### PR TITLE
ci(gitlab): try to restore once

### PR DESCRIPTION
**Pull request by douglasduteil**

### CODE REVIEW

**Comment by douglasduteil** [comment]
Demo
image

**Comment by douglasduteil** [comment]
Demo on the once 

image

File: targets/hasura/.gitlab-ci.yml [hunk 1]
```diff
@@ -23,8 +23,8 @@
   variables:
     KOSKO_GENERATE_ARGS: --env dev jobs/restore
   before_script:
-    - echo "kubectl config set-context --current --namespace=${K8S_NAMESPACE}"
-    - kubectl get job.batch/restore-db-alpha || exit 0
+    - "[[ -n $FORCE ]] && kubectl -n cdtn-admin-secret delete all -l component=${CI_COMMIT_REF_SLUG} || true"
+    - kubectl -n cdtn-admin-secret get all -l component=${CI_COMMIT_REF_SLUG} && echo "Do not recreate jobs" && exit 0
```
Comment (revolunet) [inline targets/hasura/.gitlab-ci.yml:27]: Why the 2nd instruction?

Comment (lionelB) [inline targets/hasura/.gitlab-ci.yml:27]: pour le message `skip restore db`  ?

Comment (douglasduteil) [reply targets/hasura/.gitlab-ci.yml:27]: @revolunet 

> Why the 2nd instruction?

- 1nd instruction : remove existing jobs config etc... from the cdtn-admin-secret on production if we `FORCE` the restoration
- 2nd instruction : Skip the job if found on the cdtn-admin-secret on production

**Comment by douglasduteil** [comment]
🆙 @revolunet @lionelB

File: .k8s/components/jobs/restore/container.ts [hunk 2]
```diff
@@ -18,6 +21,8 @@ const job = restoreContainerJob({
   to: "dev",
 });
 job.metadata!.name = `restore-container-${GITLAB_LIKE_ENVIRONMENT_SLUG}`;
+job.metadata!.labels = gitlabEnv.labels || {};
+job.metadata!.labels.component = `${process.env.CI_COMMIT_REF_SLUG}`;
```
Comment (revolunet) [inline .k8s/components/jobs/restore/container.ts:25]:
```suggestion
job.metadata!.labels.component = `restore-${process.env.CI_COMMIT_REF_SLUG}`;
```

File: .k8s/components/jobs/restore/db.ts [hunk 3]
```diff
@@ -34,6 +36,13 @@ const manifests = restoreDbJob({
     .toString(),
   project: "cdtn-admin",
 });
+
+manifests.forEach((m) => {
+  m.metadata = m.metadata || {};
+  m.metadata.labels = gitlabEnv.labels || {};
+  m.metadata.labels.component = `${process.env.CI_COMMIT_REF_SLUG}`;
```
Comment (revolunet) [inline .k8s/components/jobs/restore/db.ts:43]:
```suggestion
  m.metadata.labels.component = `restore-${process.env.CI_COMMIT_REF_SLUG}`;
```

File: targets/hasura/.gitlab-ci.yml [hunk 4]
```diff
@@ -23,8 +23,8 @@
   variables:
     KOSKO_GENERATE_ARGS: --env dev jobs/restore
   before_script:
-    - echo "kubectl config set-context --current --namespace=${K8S_NAMESPACE}"
-    - kubectl get job.batch/restore-db-alpha || exit 0
+    - "[[ -n $CDTN_FORCE_RESTORE_DB ]] && kubectl -n cdtn-admin-secret delete all -l component=${CI_COMMIT_REF_SLUG} || true"
```
Comment (revolunet) [inline targets/hasura/.gitlab-ci.yml:26]:
```suggestion
    - "[[ -n $CDTN_FORCE_RESTORE_DB ]] && kubectl -n cdtn-admin-secret delete all -l component=restore-${CI_COMMIT_REF_SLUG} || true"
```

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -12,13 +12,13 @@ include:
...
   - /targets/ingester-elasticsearch/.gitlab-ci.yml
 
-
 #
 #
 #
 
 # variables:
 #   AUTO_DEVOPS_ENABLE_KAPP: "🕹️"
+#   CDTN_FORCE_RESTORE_DB: "🌪️" # Will force restore the database !
```
Comment (lionelB) [inline .gitlab-ci.yml:21]:
```suggestion
   CDTN_FORCE_RESTORE_DB: 
     value: "false" 
     description: "Will force restore the database !"
```

**Review by lionelB — APPROVED** [review]

**Pull request merged**